### PR TITLE
Lc/highest lowest score

### PR DESCRIPTION
**Pull request by exasperlnc**
### Describe your changes
- add highest/lowest total score methods to stat_tracker
- add score_count method to game class
- 

### Checklist before requesting a review
- [ x] I have performed a self-review of my code.
- [ x] I have added thorough tests - unit and integrated.

### Number of teammates to review this PR before merging
- [ x] one 
- [ ] two 
- [ ] EVERYONE PLS

### CODE REVIEW

File: lib/game.rb [hunk 1]
```diff
@@ -18,4 +18,8 @@ def initialize(row)
     @away_goals = row[:away_goals].to_i
     @home_goals = row[:home_goals].to_i
   end
+
+  def score_count
```
Comment (grace-joh) [inline lib/game.rb]: Could we call this total_score to keep it consistent with the highest and lowest total score methods?

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -30,4 +30,16 @@ def all_teams
       team
     end
   end
+
+  def highest_total_score
+    all_games.map do |game|
+      game.score_count
```
Comment (grace-joh) [inline lib/stat_tracker.rb:36]:
```suggestion
      game.total_score
```

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -30,4 +30,16 @@ def all_teams
...
+    end.max
+  end
+
```
Comment (grace-joh) [inline lib/stat_tracker.rb:39]: Do you think the enumerables .max_by and .min_by would do the same thing as .map + .max/min ?

File: lib/game.rb (hunk 1 above)
Comment (exasperlnc) [inline lib/game.rb]:
```suggestion
  def total_score
```

**Review by grace-joh — CHANGES_REQUESTED** [review]
@exasperlnc see comments for fix and refactor suggestions!

File: lib/stat_tracker.rb [hunk 4]
```diff
@@ -30,4 +30,16 @@ def all_teams
...
+  def lowest_total_score
+    all_games.map do |game|
+      game.score_count
```
Comment (exasperlnc) [inline lib/stat_tracker.rb]:
```suggestion
      game.total_score
```

File: spec/game_spec.rb [hunk 5]
```diff
@@ -28,4 +28,8 @@
     expect(@game.away_goals).to eq(2)
     expect(@game.home_goals).to eq(3)
   end
+
+  it 'counts score' do
+    expect(@game.score_count).to eq(5)
```
Comment (exasperlnc) [inline spec/game_spec.rb]:
```suggestion
    expect(@game.total_score).to eq(5)
```

File: lib/stat_tracker.rb (hunk 2 above)
Comment (exasperlnc) [inline lib/stat_tracker.rb]:
```suggestion
      game.total_score
```

File: lib/stat_tracker.rb (hunk 3 above)
Comment (grace-joh) [reply lib/stat_tracker.rb:39]: @Logan and I together saw that max_by and min_by returns the object and not the score of the max/min

**Review by grace-joh — APPROVED** [review]

**Pull request merged**